File: pubnub-gson/src/main/java/com/pubnub/api/models/server/Envelope.java
package com.pubnub.api.models.server;

import com.google.gson.JsonElement;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class Envelope<T> {
    private int status;
    private String message;
    private String service;
    private T payload;
    private int occupancy;
    private JsonElement uuids;
    private String action;
    private boolean error;
}

File: core/src/main/kotlin/com/pubnub/api/models/server/Envelope.kt
package com.pubnub.api.models.server

import com.google.gson.JsonElement

class Envelope<T> {
    internal val status: Int = 0
    internal val message: String? = null
    internal val service: String? = null
    internal val payload: T? = null
    internal val occupancy: Int = 0
    internal val uuids: JsonElement? = null
    internal val action: String? = null
    internal val error: Boolean = false
}
